src/appkit.cs

```diff
@@ -27427,4 +27575,265 @@ interface NSStepperTouchBarItem
+		NSTextInputTraitType TextReplacementType { get; set; }
+
+		[Export ("dataDetectionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType DataDetectionType { get; set; }
+
+		[Export ("linkDetectionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType LinkDetectionType { get; set; }
+
+		[Export ("textCompletionType", ArgumentSemantic.Assign)]
+		NSTextInputTraitType TextCompletionType { get; set; }
+	}
+
+	interface INSTextCheckingClient { }
+
+	[Protocol]
+	interface NSTextCheckingClient : NSTextInputTraits, NSTextInputClient
+	{
+		[Abstract]
+		[Export ("annotatedSubstringForProposedRange:actualRange:")]
+		[return: NullAllowed]
+		NSAttributedString AnnotatedSubstring (NSRange range, [NullAllowed] NSRange actualRange);
+
+		[Abstract]
+		[Export ("setAnnotations:range:")]
+		void SetAnnotations (NSDictionary<NSString, NSString> annotations, NSRange range);
+
+		[Abstract]
+		[Export ("addAnnotations:range:")]
+		void AddAnnotations (NSDictionary<NSString, NSString> annotations, NSRange range);
+
+		[Abstract]
+		[Export ("removeAnnotation:range:")]
+		void RemoveAnnotation (string annotationName, NSRange range);
+
+		[Abstract]
+		[Export ("replaceCharactersInRange:withAnnotatedString:")]
+		void ReplaceCharacters (NSRange range, NSAttributedString annotatedString);
+
+		[Abstract]
+		[Export ("selectAndShowRange:")]
+		void SelectAndShow (NSRange range);
+
+		[Abstract]
+		[Export ("viewForRange:firstRect:actualRange:")]
+		[return: NullAllowed]
+		NSView GetView (NSRange range, [NullAllowed] CGRect firstRect, [NullAllowed] NSRange actualRange);
+
+		[Abstract]
+		[NullAllowed, Export ("candidateListTouchBarItem")]
+		NSCandidateListTouchBarItem CandidateListTouchBarItem { get; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof (NSObject))]
+	interface NSWorkspaceOpenConfiguration : NSCopying
+	{
+		[Static]
+		[Export ("configuration")]
+		NSWorkspaceOpenConfiguration Configuration ();
+
+		[Export ("promptsUserIfNeeded")]
+		bool PromptsUserIfNeeded { get; set; }
+
+		[Export ("addsToRecentItems")]
+		bool AddsToRecentItems { get; set; }
+
+		[Export ("activates")]
+		bool Activates { get; set; }
+
+		[Export ("hides")]
+		bool Hides { get; set; }
+
+		[Export ("hidesOthers")]
+		bool HidesOthers { get; set; }
+
+		[Export ("forPrinting")]
+		bool ForPrinting { [Bind ("isForPrinting")] get; set; }
+
+		[Export ("createsNewApplicationInstance")]
+		bool CreatesNewApplicationInstance { get; set; }
+
+		[Export ("allowsRunningApplicationSubstitution")]
+		bool AllowsRunningApplicationSubstitution { get; set; }
+
+		[Export ("arguments", ArgumentSemantic.Copy)]
+		string[] Arguments { get; set; }
+
+		[Export ("environment", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSString> Environment { get; set; }
+
+		[NullAllowed, Export ("appleEvent", ArgumentSemantic.Strong)]
+		NSAppleEventDescriptor AppleEvent { get; set; }
+
+		[Export ("architecture")]
+		int Architecture { get; set; }
```

Comment:
`cpu_type_t` -> we might have an enum that map to this

Reply:
So there is the `CFBundle.Architecture` enum that maps the `cpu_type_t`s. I'll move it to CoreFoundation_Core_Sources to make it available here. However there is a generator issue https://github.com/xamarin/maccore/issues/2041 that is causing an error right now. @mandel-macaque and I found a fix for it, so we'll get that working